addon/addon/services/animations.ts

```diff
@@ -202,7 +203,26 @@ export default class AnimationsService extends Service {
       let contextNode = this.spriteTree.lookupNodeByElement(
         context.element
       ) as SpriteTreeNode;
-      contextNode.freshlyRemovedChildren.clear();
+      for (let item of contextNode.freshlyRemovedChildren) {
+        if (item.isSprite) {
+          let identifier = new SpriteIdentifier(
+            (item.spriteModel as SpriteStateTracker).id,
+            (item.spriteModel as SpriteStateTracker).role
+          ).toString();
+          if (
+            !(
+              context.orphans.has(identifier) &&
+              this.intermediateSprites.get(identifier)
+            )
+          ) {
+            (item.parent as SpriteTreeNode).freshlyRemovedChildren.delete(item);
+          } else {
+            this.interruptedRemoved.add(item.spriteModel as SpriteStateTracker);
```

Comment:
here we are leaving the sprite's node in `freshlyRemovedChildren`. This currently creates a bug when interrupting rapidly:
- The sprite is a removed sprite, and in its controlling context's `freshlyRemovedChildren`
- The sprite is interrupted and becomes a kept sprite. In the process we don't currently remove it from `freshlyRemovedChildren`. 
- The new sprite is interrupted and becomes a removed sprite and is added to `freshlyRemovedChildren`. We now have 2 copies of this sprite in `freshlyRemovedChildren`.

Reply:
To perform cleanup for this properly, we need to guarantee that we can identify a sprite as matching another one, which needs the id property to be compulsory... We are using that property in a bunch of places to match things with sprite modifiers or sprites.

Reply:
think about situations where we want to clean these up, and situations where we won't